File: 2658/2658.c
int maximumNumberOfFishInAGrid(struct TreeNode* root) {
    if (!root) return 0;
    int l = maximumNumberOfFishInAGrid(root->left);
    int r = maximumNumberOfFishInAGrid(root->right);
    return 1 + (l > r ? l : r);
}

File: 2658/2658.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maximumNumberOfFishInAGrid = function(root) {
    if (!root) return 0;
    return 1 + Math.max(maximumNumberOfFishInAGrid(root.left), maximumNumberOfFishInAGrid(root.right));
};